application/ui/src/features/inspect/jobs/show-job-logs.component.tsx

```diff
@@ -0,0 +1,129 @@
+import { Suspense } from 'react';
+
+import {
+    ActionButton,
+    Button,
+    ButtonGroup,
+    Content,
+    Dialog,
+    DialogTrigger,
+    Divider,
+    Flex,
+    Heading,
+    Icon,
+    Loading,
+    Text,
+    View,
+} from '@geti/ui';
+import { LogsIcon } from '@geti/ui/icons';
+import { queryOptions, experimental_streamedQuery as streamedQuery, useQuery } from '@tanstack/react-query';
+
+// Connect to an SSE endpoint and yield its messages
+function fetchSSE(url: string) {
+    return {
+        async *[Symbol.asyncIterator]() {
+            const eventSource = new EventSource(url);
+
+            try {
+                let { promise, resolve, reject } = Promise.withResolvers<string>();
+
+                eventSource.onmessage = (event) => {
+                    if (event.data === 'DONE' || event.data.includes('COMPLETED')) {
+                        eventSource.close();
+                        resolve('DONE');
+                        return;
+                    }
+                    resolve(event.data);
+                };
+
+                eventSource.onerror = (error) => {
+                    eventSource.close();
+                    reject(new Error('EventSource failed: ' + error));
+                };
+
+                // Keep yielding data as it comes in
+                while (true) {
+                    const message = await promise;
+
+                    // If server sends 'DONE' message or similar, break the loop
+                    if (message === 'DONE') {
+                        break;
+                    }
+
+                    try {
+                        const data = JSON.parse(message);
+                        if (data['text']) {
+                            yield data['text'];
+                        }
+                    } catch {
+                        console.error('Could not parse message:', message);
+                    }
+
+                    ({ promise, resolve, reject } = Promise.withResolvers<string>());
```

Comment:
Nice!!!

Reply:
First time using this method, felt like a nice use case. :D 
I did worry a bit that maybe this code is a bit too weird, but I expect we'd only need it here so I don't see too many issues with it.